src/parcsr_ls/par_mgr_device.c

```diff
@@ -692,27 +692,27 @@ hypre_ParCSRMatrixExtractBlockDiagDevice( hypre_ParCSRMatrix   *A,
 
       /* Memory allocation */
       tmpdiag     = hypre_TAlloc(HYPRE_Complex, bdiag_size, HYPRE_MEMORY_DEVICE);
-      diag_aop    = hypre_TAlloc(HYPRE_Complex *, num_rows, HYPRE_MEMORY_DEVICE);
+      diag_aop    = hypre_TAlloc(HYPRE_Complex *, num_blocks, HYPRE_MEMORY_DEVICE);
```

Comment:
num_blocks is used to replace num_rows in various places. Was this a bug before, or were we allocating and using more memory than was needed?

Reply:
I was allocating more info than needed